### DISCUSSION TITLE
Database Restore Best Practices

### DISCUSSION THREAD

**Discussion by jon-m-bottoms** [ravendb/ravendb#18437] [Q&A] [answered]
We keep regular full and incremental backups and are able to restore. However, there only option seems to be to restore to a NEW database. 

If we have a production issue, say a bad patch deletes or corrupts are large number of documents. I would like to quickly restore the current database from the last known good backup. What is the recommended approach here?

**Comment by ayende** [comment]
That would basically be:
* restore as a new db
* rename the database

You want to restore on top of the existing database? What would be the behavior of existing values that aren't in the backup, for example?
What is the state of ETL or subscriptions, etc?

In short, that isn't valid.

Note that the backup format is the same as the export format. You _can_ import data to an existing database.
It will overwrite whatever you have already, and won't touch documents that have been added since the backup.

Separately, we have a couple feature specifically for your scenario:

* Off site backups using delayed replication. Meaning that modifications in the database take time to replicate to the other side.
* Revert Revisions  - uses the revisions feature to go back to an older point in time.
 
![image](https://github.com/ravendb/ravendb/assets/116915/5fb31253-5a39-4241-807e-4e1c9fa287b0)

**Comment by jon-m-bottoms** [comment]
Fair enough. How do I rename a database?

**Reply by ayende** [reply, marked as answer]
The easiest way it to do a soft delete and create a new database with the new name and old path.

**Reply by ayende** [reply]
FWIW, see: https://ravendb.net/learn/inside-ravendb-book/reader/4.0/18-operational-recipes#renaming-a-database

Basically, that is a problematic situation on actual rename. 

See more details here: 
https://ravendb.net/learn/inside-ravendb-book/reader/4.0/17-backups-and-restores#restoring-to-multiple-nodes-at-the-same-time
